analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -130,142 +116,161 @@ Because `DBI` does not support full_join, we had to use a series of `left_join`
+          plot_file_path = file.path(plots_dir, "pbta-upset-plot-transcipts-only.png"), 
+          subset_vector = transcripts)
 ```
 
+![Upset plot - transcripts only](plots/pbta-caller-comparison/pbta-upset-plot-transcipts-only.png)
+
 Non transcript upset plot. 
 
 ```{r}
-callers_per_mutation %>% 
-  dplyr::filter(Variant_Classification %in% c("5'Flank", "3'Flank", "IGR", "Intron")) %>%
-  upset_plot()
+not_transcripts <- all_caller_df$Variant_Classification %in% c("5'Flank", "3'Flank", "IGR", "Intron") 
+
+upset_png(detect_mat,
+          plot_file_path = file.path(plots_dir, "pbta-upset-plot-not-transcipts-only.png"), 
+          subset_vector = transcripts)
 ```
 
-## VAF for combination sets of callers
+![Upset plot - transcripts only](plots/pbta-caller-comparison/pbta-upset-plot-not-transcipts-only.png)
+
+## VAF distribution for each caller
 
-Graph mutations by combinations of callers and plot the VAF density.
-Get list of callers per mutation and the median vaf for each. 
-Graph the median VAF for each combination of callers.
+Create consensus VAF based on strelka's VAFs and which mutations would be included
+
+```{r}
+# Make new VAF column based on strelka's VAFs
+all_caller_df$VAF_consensus <- all_caller_df$VAF_strelka
+
+# Figure out which mutations are not in the consensus
+not_consensus <- dimnames(consensus_count)[[1]][which(consensus_count < 3)]
+
+# Make non consensus mutations in consensus)VAF an NA
+all_caller_df$VAF_consensus[not_consensus] <- NA
+```
+
+Make a long form VAF data.frame for exploring mutations VAFs from each caller. 
+
+```{r}
+vaf_df <- all_caller_df %>% 
+  dplyr::select(index, dplyr::starts_with("VAF_"), Variant_Classification) %>%
+  # Make long format
+  tidyr::gather(key = "caller", value = "vaf", -index, -Variant_Classification) %>% 
+  # Drop the `VAF_`
+  dplyr::mutate(caller = gsub("VAF_", "", caller)) %>% 
+  dplyr::filter(!is.na(vaf))
+```
 
 ```{r}
 # Make this plot
-median_vaf_plot <- callers_per_mutation %>%
-  ggplot2::ggplot(ggplot2::aes(x = caller_combo, y = median_vaf)) +
+vaf_plot <- vaf_df %>%
+  ggplot2::ggplot(ggplot2::aes(x = caller, y = vaf, color = caller)) +
   ggplot2::geom_violin() +
   ggplot2::theme_classic() +
-  ggupset::scale_x_mergelist(sep = "-") +
-  ggupset::axis_combmatrix(sep = "-") +
   ggplot2::xlab("") +
-  ggplot2::ylab("Median VAF Across Callers")
+  ggplot2::ylab("VAF") + 
+  ggplot2::theme(legend.position = "none")
 
 # Print out
-median_vaf_plot
+vaf_plot
 ```
 
 ```{r}
 # Save this plot
-ggplot2::ggsave(plot = median_vaf_plot, 
-                file.path(plots_dir, "upset_median_vaf_plot.png"))
+ggplot2::ggsave(plot = vaf_plot, 
+                file.path(plots_dir, "pbta-vaf_distribution_plot.png"))
 ```
 
-## Base Changes 
+## VAF correlations
+
+Correlate VAF's across callers.
+
+```{r}
+cor_vaf <- all_caller_df %>% 
+  dplyr::select(dplyr::starts_with("VAF_"), -VAF_consensus) %>% 
+  GGally::ggpairs(mapping = ggplot2::aes(alpha = 0.05)) 
```

Comment:
`alpha` seems like it is being ignored here, or not getting its full due and I don't know enough to know why. Probably not worth worrying about.

Reply:
🤷‍♀️ It does work if you look at the very fringes it just doesn't apply it to the lines only the fillings, and for PBTA there's just so many plots it doesn't matter anyway.